yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -60,6 +60,7 @@ private[spark] class Client(
 
   def this(clientArgs: ClientArguments) = this(clientArgs, new SparkConf())
 
+  private var stopMonitor = false
```

Comment:
This would have to be `@volatile`. The problem with this construction is that several callers to the `monitorApplication` method will now collide, so I'm not sure this is safe.